Sources/Core/Views/TextField/TextField.swift

```diff
@@ -0,0 +1,273 @@
+//
+//  TextField.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 01/20/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import UIKit
+
+public protocol RuleValidatable {
+  var error: String { get }
+  func validate(_ input: String?) -> Bool
+}
+
+public class TextField: UIView {
+  private let stackView = UIStackView()
+  private let titleLabel = PaddingLabel()
+  private let valueContainerView = UIView()
+  private let valueTextField = UITextField()
+  private let errorLabel = UILabel()
+  private var rule: RuleValidatable?
+  private var textFieldHeightAnchor: NSLayoutConstraint?
+
+  weak public var delegate: UITextFieldDelegate? {
+    didSet { valueTextField.delegate = delegate }
+  }
+  
+  public init(with rule: RuleValidatable? = .none) {
```

Comment:
Maybe it would be also good to have option to set rule properly separate from init

`func setRule(_ rule: RuleValidatable) {`